app/s3/S3.scala

```diff
@@ -5,38 +5,54 @@ import com.amazonaws.services.s3.model._
 import config._
 import helpers.Loggable
 import models.{Snapshot, SnapshotId}
-import play.api.data.validation.ValidationError
-import play.api.libs.json.{JsPath, Json}
+import play.api.libs.json.{JsValue, Json}
 
 import scala.collection.JavaConverters._
 import scala.io.Source
 
 class S3(config: RestorerConfig, s3Client: AmazonS3Client) extends Loggable {
   lazy val snapshotBucket: String = config.snapshotBucket
 
-  def getRawSnapshot(snapshotId: SnapshotId): String = getObjectContents(snapshotId.key, snapshotBucket)
+  def getRawSnapshot(snapshotId: SnapshotId): Either[String, String] = getObjectContentRaw(snapshotId.key, snapshotBucket)
 
-  def getSnapshot(snapshotId: SnapshotId): Either[Seq[(JsPath, Seq[ValidationError])], Snapshot] = {
-    val obj = getObject(snapshotId.key, snapshotBucket)
-    try {
-      val json = Json.parse(obj.getObjectContent)
-      Json.fromJson[Snapshot](json).asEither
-    } finally {
-      obj.close()
+  def getSnapshot(snapshotId: SnapshotId): Either[String, Snapshot] = {
+    getObjectContentJson(snapshotId.key, snapshotBucket).right.map { json =>
+      Snapshot(snapshotId, json)
     }
   }
 
-  private def getObject(key: String, bucketName: String): S3Object = {
-    s3Client.getObject(new GetObjectRequest(bucketName, key))
+  def getRawSnapshotInfo(snapshotId: SnapshotId): Either[String, String] = getObjectContentRaw(snapshotId.infoKey, snapshotBucket)
+
+  def getSnapshotInfo(snapshotId: SnapshotId): Either[String, JsValue] =
+    getObjectContentJson(snapshotId.infoKey, snapshotBucket)
+
+  private def getObject(key: String, bucketName: String): Either[String, S3Object] = {
+    try {
+      Right(s3Client.getObject(new GetObjectRequest(bucketName, key)))
+    } catch {
+      case e:AmazonS3Exception if e.getErrorCode == "NoSuchKey" =>
+        Left("Object doesn't exist")
```

Comment:
Do we need to handle any other exceptions including AmazonS3Exception without error code of `NoSuchKey`?